### PR TITLE
Add enabled variable and encrypt CloudWatch log group

### PR DESCRIPTION
**Pull request by pjdufour-dds**
👋 This PR adds a variable, so one could easily "pause" cloudtrail logging without having to comment out the whole module.  This PR also adds a variable to encrypt the logs when stored in a cloudwatch log group.

### CODE REVIEW

**Comment by esacteksab** [comment]
Pulled the branch, tests passed

```
PASS
ok  	github.com/trussworks/terraform-aws-cloudtrail/test	85.044s
```

**Comment by eeeady** [comment]
I'm having trouble with tests... might be my local environment then.

**Comment by esacteksab** [comment]
> I'm having trouble with tests... might be my local environment then.

Got a snippet?

**Comment by pjdufour-dds** [comment]
I split the enabled flag into a separate PR from the encryption here.  https://github.com/trussworks/terraform-aws-cloudtrail/pull/24. Once the enabled PR is merged, I'll resync this PR.

**Comment by pjdufour-dds** [comment]
Resynced with master.

File: variables.tf [hunk 1]
```diff
@@ -50,3 +44,15 @@ variable "s3_key_prefix" {
   default     = "cloudtrail"
   type        = string
 }
+
+variable "cloudwatch_log_group_retention_in_days" {
```
Comment (dynamike) [inline variables.tf]:
```suggestion
variable "cloudwatch_log_retention_in_days" {
```

File: variables.tf [hunk 2]
```diff
@@ -50,3 +44,15 @@ variable "s3_key_prefix" {
...
+  description = "Number of days to keep the CloudTrail logs in the CloudWatch log group."
+  default     = 90
+  type        = string
```
Comment (dynamike) [inline variables.tf]:
```suggestion
  type        = number
```

**Comment by pjdufour-dds** [comment]
Given that the key policy used for encrypting CloudTrail files delivered to S3 is really complex, it makes a lot of sense to manage that key policy in this module.

For CloudWatch exports, I'd like to use one `alias/cloudwatch` KMS key with a simple policy to encrypt all CloudWatch log groups for an AWS account, which is why I was parameterizing with this variable.  When using a KMS key managed outside the module, as in this PR, I can easily rotate all my CloudWatch log groups across projects to a new key all at once.

**Comment by dynamike** [comment]
Gonna close this. We'll be implementing encryption within the module in the coming weeks

**Pull request closed without merging**